src/env-inl.h

```diff
@@ -613,7 +613,7 @@ void Environment::RequestInterrupt(Fn&& cb) {
 }
 
 inline bool Environment::can_call_into_js() const {
-  return can_call_into_js_ && !is_stopping();
+  return can_call_into_js_;
```

Comment:
If we do this, we should add something like `DCHECK_IMPLIES(is_stopping(), !can_call_into_js_);` to ensure that the semantics are still correct here.

Reply:
Thanks for the suggestion! Updated.

Reply:
This invariant here is being broken in `ExitEnv()` now, though, right?

You could probably work around that by putting the entire body of `ExitEnv()` into a `RequestInterrupt()` instead of the `SetImmediateThreadsafe()` but honestly, I would just recommend leaving this function here as it was previously.

Reply:
Yeah, right. Thanks for pointing this out.